tc/core/polyhedral/scop.h

```diff
@@ -229,6 +245,20 @@ struct Scop {
     return true;
   }
 
+  static bool isWarpSyncId(isl::id id) {
+    auto name = id.get_name();
```

Comment:
I would abstract this and `isSyncId` away to a function that takes an extra string argument of expected id prefix.